dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventChangeLogService.java

```diff
@@ -55,16 +56,29 @@ Page<EventChangeLog> getEventChangeLog(
   List<TrackedEntityDataValueChangeLog> getTrackedEntityDataValueChangeLogs(
       TrackedEntityDataValueChangeLogQueryParams params);
 
+  @Deprecated(since = "2.42")
   void addTrackedEntityDataValueChangeLog(
       TrackedEntityDataValueChangeLog trackedEntityDataValueChangeLog);
 
+  void addDataValueChangeLog(
+      Event event,
+      DataElement dataElement,
+      String currentValue,
```

Comment:
Maybe not so important but I would expect previous to go before than current as it is done in the view object.